pkg/flow/adapter/xmltojsontransformation/adapter.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2022 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package xmltojsontransformation
+
+import (
+	"context"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"go.uber.org/zap"
+
+	xj "github.com/basgys/goxml2json"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	pkgadapter "knative.dev/eventing/pkg/adapter/v2"
+	"knative.dev/pkg/logging"
+
+	"github.com/triggermesh/triggermesh/pkg/apis/flow/v1alpha1"
+	targetce "github.com/triggermesh/triggermesh/pkg/targets/adapter/cloudevents"
+)
+
+// EnvAccessorCtor for configuration parameters
+func EnvAccessorCtor() pkgadapter.EnvConfigAccessor {
+	return &envAccessor{}
+}
+
+type envAccessor struct {
+	pkgadapter.EnvConfig
+
+	// BridgeIdentifier is the name of the bridge workflow this target is part of
+	BridgeIdentifier string `envconfig:"EVENTS_BRIDGE_IDENTIFIER"`
+	// CloudEvents responses parametrization
+	CloudEventPayloadPolicy string `envconfig:"EVENTS_PAYLOAD_POLICY" default:"error"`
+}
+
+// NewTarget adapter implementation
+func NewTarget(ctx context.Context, envAcc pkgadapter.EnvConfigAccessor, ceClient cloudevents.Client) pkgadapter.Adapter {
```

Comment:
`NewAdapter`.
What you're creating is an adapter (doesn't matter what object it is backing, still an adapter in the Knative sense). "Target" is the type of object, which isn't even true here.